jupyter_kernel_mgmt/subproc/manager.py

```diff
@@ -127,9 +117,10 @@ def signal(self, signum):
                 return
             except OSError:
                 pass
-        self.kernel.send_signal(signum)
+        await maybe_future(self.kernel.send_signal(signum))
 
-    def is_alive(self):
+    async def is_alive(self):
         """Is the kernel process still running?"""
-        return self.kernel.poll() is None
+        poll_result = await maybe_future(self.kernel.poll())
```

Comment:
asyncio `Process` objects don't have a `poll()` method. It looks like we're still using `Popen` at the moment, but I guess the idea is to move to asyncio to allow async `wait()`?

Reply:
This repo still _has_ both.  I wasn't comfortable removing the synchronous form until the BlockingClient stuff is ferreted out (e.g., test_client).  I'm hoping you can make the appropriate changes/decisions for that.

Reply:
The process management part should be separate from the client implementation - I think you can switch to asyncio for the former without worrying about the latter. It doesn't have to be in this PR, but let's not forget that we'll need to replace `poll()` when we do.

Reply:
Sorry Thomas.  Its not clear to me what you're proposing I do. Are you suggesting we keep the `KernelManager` _name_, but essentially use the `AsyncPopenKernelManager` _content_ its its place (filling calls to `super()...` with the appropriate code from `KernelManager`)?

Reply:
Yes, I think it's better to go async where possible and make sync wrappers around that.

Reply:
I see, I already made `BlockingKernelClient` a wrapper around the async implementation (IOLoopKernelClient). I think that can stay as it is, then.

I think the issue is what to do about the high-level convenient functions `start_new_kernel` and `run_kernel`, right? Maybe they need to be rethought - they're inherited from jupyter_client and still expect to be given an explicit argv. I think it would still be useful to have a high-level "give me a running kernel" API like that, but perhaps it should take a kernel type ID rather than argv.

Can you see a way to pass over them for this PR (which is already big enough) so we can tackle that as a separate step? Even if we xfail some tests for now until we rethink them.

Reply:
I agree on this PR's size.  Yes, `run_kernel`, doesn't have a presence the async side because async context managers are problematic (I think perhaps only pre 3.6 IIRC).

Before I continue - what are your thoughts on moving AsyncPopenKernelManager directly to KernelManager and getting rid of `async_manager.py` (so same for the launchers)?